pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -541,6 +550,10 @@ func (r *ReconcileClusterSync) applySyncSets(
 				// or SyncSet creationTimestamp.
 				logger.WithField("applyTime", applyTime).Debug("observed first successful apply of SyncSet for cluster")
 				metricTimeToApplySyncSet.Observe(applyTime)
+
+				if syncSetGroup, ok := syncSet.AsMetaObject().GetAnnotations()[constants.SyncSetMetricsGroupAnnotation]; ok {
+					metricTimeToApplyGroupedSyncSet.WithLabelValues(syncSetGroup).Observe(applyTime)
+				}
```

Comment:
My vote would be to keep them reported in metricTimeToApplySyncSet, that should be over all SyncSets and it would get hard to explain if it were just those without an explicit grouping.

I really like your idea of combining them to avoid repetition. Lets add the group label to metricTimeToApplySyncSet. I think prometheus would require us to set some value for those without a group, maybe "none" or something like that?

Reply:
Yes - grouping sounds the most ideal. We could label the syncsets without a metrics label as "none".
From a statistical perspective, it makes sense to keep them clubbed under 1 graph - so we can simultaneously see the stats for the grouped SyncSets as well as apply aggregation to derive the results for all SyncSets.
@akhil-rane does this seem okay to you?

Reply:
looks good to me